qiskit/dagcircuit/dagdependency.py

```diff
@@ -461,14 +461,27 @@ def _update_edges(self):
             if reachable[prev_node_id]:
                 prev_node = self.get_node(prev_node_id)
 
-                if not self.comm_checker.commute(
-                    prev_node.op,
-                    prev_node.qargs,
-                    prev_node.cargs,
-                    max_node.op,
-                    max_node.qargs,
-                    max_node.cargs,
+                # TODO For some reason, DAGDependency needs gates *not* to commute if they are
```

Comment:
Before this change, DAGDependency used to look like (and still does with this hacky if-statement)
<img width="198" alt="image" src="https://github.com/user-attachments/assets/df60b874-c54f-471a-aa29-066b040c8728">

but since P and RZ do commute, it would actually look like this
<img width="731" alt="image" src="https://github.com/user-attachments/assets/7843ee43-664a-483c-9e77-bd3c1063a610">

However, correctly taking into account these commutations breaks the template matching -- it seems like this is something the template matching should be able to handle, but for now I added this flag.